src/documents/to-do/DeveloperIntroductionToInfusionFramework.md

```diff
@@ -4,6 +4,595 @@ layout: default
+        sayHello: "fluid.notImplemented"
+    }
+});
+
+fluid.defaults("fluid.helloWorld.consoleHello", {
+    // This component has all of the characteristics of sayHello,
+    // except for its implementation in the invoker
+    gradeNames: ["fluid.helloWorld.sayHello"],
+    invokers: {
+        sayHello: {
+            "this": "console",
+            "method": "log",
+            // Here, "{that}" means the context of the current
+            // component configuration (consoleHello)
+            "args": ["{that}.model.message"]
+        },
+    }
+});
+
+fluid.defaults("fluid.helloWorld.displayHello", {
+    // This component has all of the characteristics of sayHello,
+    // except for its implementation in the invoker
+    gradeNames: ["fluid.helloWorld.sayHello"],
+    invokers: {
+        sayHello: {
+            "this": "{helloWorld}.dom.messageArea",
+            "method": "html",
+            "args": ["{that}.model.message"]
+        }
+    }
+});
+
+fluid.defaults("fluid.helloWorld.speakHello", {
+    // This component has all of the characteristics of sayHello,
+    // except for its implementation in the invoker
+    // We also "mix in" the fluid.textToSpeech component to give it
+    // the capability to access the browser's text to speech interface
+    gradeNames: ["fluid.textToSpeech", "fluid.helloWorld.sayHello"],
+    invokers: {
+        sayHello: {
+            // This style of Invoker allows us to refer to another
+            // existing invoker using IoC references - in this case the
+            // "queueSpeech" invoker that we have access to from mixing
+            // in the fluid.textToSpeech grade
+            "func": "{that}.queueSpeech",
+            "args": ["{that}.model.message"]
+        }
+    }
+});
+
+fluid.defaults("fluid.helloWorld", {
+    gradeNames: ["fluid.viewComponent"],
+    model: {
+        message: "Hello, World!"
+    },
+    selectors: {
+        messageArea: ".flc-messageArea"
+    },
+    listeners: {
+        "onCreate.announceSelf": {
+            "this": "console",
+            "method": "log",
+            "args": ["The helloWorld Component is ready"]
+        }
+    },
+    components: {
+        consoleHello: {
+            type: "fluid.helloWorld.consoleHello"
+        },
+        displayHello: {
+            type: "fluid.helloWorld.displayHello",
+        },
+        speakHello: {
+            type: "fluid.helloWorld.speakHello",
+        }
+    }
+});
+```
+
+### Summing Up
+
+This elaborate "Hello, World!" implementation has slowly introduced some of Infusion's core features. To summarize:
+
+* We implement our programs by designing [components](#components) that work together to provide the needed functionality. Components are Javascript objects adhering to certain syntax conventions that are created using the `fluid.defaults` function. While components have defaults, any of these defaults can be overriden at the time a specific instance of the component is created.
```

Comment:
As I mentioned in my global comment, if you break things down by sections, you can expand on each concept within the section, and then use a condensed version here, linked to the expanded definition.